sdk/core/src/lib.rs

```diff
@@ -320,6 +320,15 @@ pub trait UserAgentSupport<'a> {
     fn with_user_agent(self, user_agent: &'a str) -> Self::O;
 }
 
+#[derive(Debug, Clone, Copy)]
+pub struct UserAgent<'a>(pub &'a str);
+
+impl<'a> AddAsHeader for UserAgent<'a> {
+    fn add_as_header(&self, builder: Builder) -> Builder {
+        builder.header(USER_AGENT, self.0)
```

Comment:
Could do this later, but this should only be additional data added to the header. See https://github.com/Azure/azure-sdk-for-net/blob/29a6bd3ae21dc01b9846b3c54cfcdb382cbc401d/sdk/core/Azure.Core/src/Pipeline/Internal/TelemetryPolicy.cs#L14-L22 for how we format this to show up in standardized reports up to our CVP.